app/controllers/application_controller.rb

```diff
@@ -1,3 +1,14 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_filter :store_current_location, :unless => :devise_controller?
+
+  private
+  def store_current_location
+    store_location_for(:user, request.url)
+  end
+
+  def after_sign_out_path_for(resource)
+    request.referrer || root_path
+  end
```

Comment:
Not sure if the main controller is the right place for this method. Also, why is it part of this PR, can't really see in what way it fits into the scope of this feature to handle the after sign out path for a user.